runtime-sdk/src/modules/core/mod.rs

```diff
@@ -872,6 +872,15 @@ impl<Cfg: Config> module::TransactionHandler for Module<Cfg> {
             return Err(Error::GasOverflow);
         }
 
+        // Attempt to limit the maximum number of consensus messages.
+        let consensus_messages = ctx.tx_auth_info().fee.consensus_messages;
+        ctx.limit_max_messages(consensus_messages)?;
+
+        // Skip additional checks/gas payment for internally generated transactions.
+        if ctx.is_internal() && !ctx.is_simulation() {
```

Comment:
if I were to do a simulation of a contract call that did a subcall, does this mean we'll additionally check the gas price on that subcall?